scripts/enforce_imports.py

```diff
@@ -0,0 +1,249 @@
+
+        # A downloaded external package
+        >>> ModuleType.from_module_name('boto3')
+        <ModuleType.external_dependency: 2>
+
+        # An internal module
+        >>> ModuleType.from_module_name('azul.indexer')
+        <ModuleType.internal: 3>
+
+        # A namespace module
+        >>> ModuleType.from_module_name('lambdas.service')
+        <ModuleType.internal: 3>
+        """
+        spec = importlib.util.find_spec(module_name)
+        if spec.origin is None:
+            # Not every module included in the Python standard library is
+            # considered "builtin." That term is reserved for modules such as
+            # `sys` that are implemented in C and lack an associated .py file.
+            assert module_name in sys.builtin_module_names
+            return cls.python_runtime
+        else:
+            if spec.origin == 'namespace':
+                # Namespace modules lack an __init__.py and represent their
+                # location as a list of diretcories.
+                path = one(importlib.import_module(module_name).__path__)
+            else:
+                path = spec.origin
+            if path.startswith(config.project_root):
+                # This assumes that external packages are installed within the
+                # virtual environment.
+                if 'site-packages' in path:
+                    return cls.external_dependency
+                else:
+                    return cls.internal
+            else:
+                # Assume any non-builtin module imported from outside the
+                # project is part of the standard library. Python installation
+                # locations can vary.
+                return cls.python_runtime
+
+
+# Module type, module name, module attribute ('' when not using from ... import)
+ModuleOrderInfo = Tuple[ModuleType, str, str]
+
+# Line, column, message
+ErrorInfo = Tuple[int, int, str]
+
+
+class ImportVisitor(ast.NodeVisitor):
+
+    def __init__(self, lines: List[str]):
+        super().__init__()
+        self.lines = lines
+        self.errors: List[ErrorInfo] = []
+        self.visited: List[Tuple[ast.AST, ModuleOrderInfo]] = []
+
+    def visit_Import(self, node: ast.Import) -> None:
+        self._check_split_import(node)
+        self._check_order(node)
+
+    def visit_ImportFrom(self, node: ast.ImportFrom) -> None:
+        self._check_split_import(node)
+        self._check_no_parentheses(node)
+        self._check_order(node)
+
+    def _check_split_import(self, node: ast.AST) -> None:
+        if len(node.names) > 1:
+            self._error(node, Errors.import_not_split)
+
+    def _check_no_parentheses(self, node: ast.AST) -> None:
+        if any(token.string == '(' for token in self._tokenize_line(node.lineno)):
+            self._error(node, Errors.import_with_parentheses)
+
+    def _check_order(self, node: ast.AST) -> None:
+        order_info = self._order_info(node)
+        if order_info is None:
```

Comment:
I just noticed this is another "bail out" as mentioned in the contributing guide.